From 25b6fb9c87e33eb83d743878be7237e47654008d Mon Sep 17 00:00:00 2001
From: Yinjun Chen <a7301c@motorola.com>
Date: Thu, 27 Aug 2015 19:38:14 -0500
Subject: [PATCH 281/959] IKQCOMSD1-2925: power: qpnp-smbcharger-mmi: Set
 DPF_DMF Prop instead of ALLOW_DETECTION

Set DPF_DMF property instead of ALLOW_DETECTION due to the following qcom commit:
1b7ff05 power: rename ALLOW_DETECTION to DPF_DMF

This change replaced the moto commit:
d63c157	IKSWL-16678: power: qpnp-smbcharger-mmi: Set Allow Detection Property

Change-Id: Ibfa31c9cc8e230966351ee38ed5e2b88db000ffd
Signed-off-by: Yinjun Chen <a7301c@motorola.com>
---
 drivers/power/qpnp-smbcharger-mmi.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/power/qpnp-smbcharger-mmi.c b/drivers/power/qpnp-smbcharger-mmi.c
index 70a1b2c9898..bf238f0afd2 100644
--- a/drivers/power/qpnp-smbcharger-mmi.c
+++ b/drivers/power/qpnp-smbcharger-mmi.c
@@ -4439,6 +4439,9 @@ static void handle_usb_removal(struct smbchg_chip *chip)
 		power_supply_set_supply_type(chip->usb_psy,
 				POWER_SUPPLY_TYPE_UNKNOWN);
 		power_supply_set_present(chip->usb_psy, chip->usb_present);
+		pr_smb(PR_MISC, "setting usb psy dp=r dm=r\n");
+		power_supply_set_dp_dm(chip->usb_psy,
+				POWER_SUPPLY_DP_DM_DPR_DMR);
 		schedule_work(&chip->usb_set_online_work);
 		rc = power_supply_set_health_state(chip->usb_psy,
 				POWER_SUPPLY_HEALTH_UNKNOWN);
@@ -4486,6 +4489,12 @@ static void handle_usb_insertion(struct smbchg_chip *chip)
 	usb_type_name = get_usb_type_name(type);
 	usb_supply_type = get_usb_supply_type(type);
 
+	if (chip->usb_psy) {
+		pr_smb(PR_MISC, "setting usb psy dp=f dm=f\n");
+		power_supply_set_dp_dm(chip->usb_psy,
+				POWER_SUPPLY_DP_DM_DPF_DMF);
+	}
+
 	/* Rerun APSD 1 sec later */
 	if ((usb_supply_type == POWER_SUPPLY_TYPE_USB) &&
 	    !chip->apsd_rerun_cnt && !chip->factory_mode) {
-- 
2.11.0

